src/run/runner/wall_time/executor.rs

```diff
@@ -38,10 +60,65 @@ impl Executor for WallTimeExecutor {
             cmd.current_dir(abs_cwd);
         }
 
-        cmd.args(["-c", get_bench_command(config)?.as_str()]);
+        // Create the perf FIFOs before starting the command
+        let mut perf_fifo = PerfFifo::new();
+        let mut ctl_fifo = FifoIpc::create(RUNNER_CTL_FIFO)
+            .unwrap()
+            .with_reader()
+            .unwrap();
 
+        // We have to pass a file to perf, which will create `perf.data.<timestamp>` files
+        // when the output is split.
+        let perf_file = tempfile::Builder::new()
+            .keep(true)
+            .prefix(PERF_DATA_PREFIX)
+            .tempfile_in(&self.perf_dir)?;
+
+        cmd.args([
+            "-c",
+            &format!(
+                "perf record --data --freq=1000 --switch-output --control=fifo:{},{} --delay=-1 -g --output={} -- {}",
+                PERF_CTL_FIFO,
+                PERF_CTL_ACK_FIFO,
+                perf_file.path().to_string_lossy(),
+                get_bench_command(config)?.as_str()
+            ),
+        ]);
         debug!("cmd: {:?}", cmd);
-        let status = run_command_with_log_pipe(cmd)
+
+        let on_process_started = |perf_pid: u32| {
+            use codspeed::fifo::Command as FifoCommand;
+
+            let mut ack_fifo = FifoIpc::create(RUNNER_ACK_FIFO)
+                .unwrap()
+                .with_reader() // FIFO needs a reader to be opened with writer
+                .unwrap()
+                .with_writer()
+                .unwrap();
+
+            debug!("Perf PID: {}", perf_pid);
+            std::thread::spawn(move || loop {
+                let Some(cmd) = ctl_fifo.recv_cmd() else {
+                    continue;
+                };
+
+                match cmd {
+                    FifoCommand::StartBenchmark => {
+                        unsafe { libc::kill(perf_pid as i32, libc::SIGUSR2) };
+                        perf_fifo.start_events();
+                        ack_fifo.send_cmd(FifoCommand::Ack).unwrap();
+                    }
+                    FifoCommand::StopBenchmark => {
+                        perf_fifo.stop_events();
+                        ack_fifo.send_cmd(FifoCommand::Ack).unwrap();
+                    }
+                    FifoCommand::Ack => unreachable!(),
+                }
+
+                std::thread::sleep(std::time::Duration::from_millis(10));
+            });
+        };
```

Comment:
in the same way, this could be an async task instead of a thread; it would avoid having to rely on polling